src/redisearch_rs/inverted_index/tests/raw_doc_ids_only.rs

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use std::io::Cursor;
+
+use ffi::{RSQueryTerm, RSTermRecord};
+use inverted_index::{Decoder, Encoder, RSIndexResult, raw_doc_ids_only::RawDocIdsOnly};
+
+#[unsafe(no_mangle)]
+pub extern "C" fn ResultMetrics_Free(result: *mut inverted_index::RSIndexResult) {
+    if result.is_null() {
+        panic!("did not expect `RSIndexResult` to be null");
+    }
+
+    let metrics = unsafe { (*result).metrics };
+    if metrics.is_null() {
+        return;
+    }
+
+    panic!(
+        "did not expect any test to set metrics, but got: {:?}",
+        unsafe { *metrics }
+    );
+}
+
+#[unsafe(no_mangle)]
+pub extern "C" fn Term_Offset_Data_Free(_tr: *mut RSTermRecord) {
+    panic!("Nothing should have copied the term record to require this call");
+}
+
+#[unsafe(no_mangle)]
+pub extern "C" fn Term_Free(_t: *mut RSQueryTerm) {
+    // The RSQueryTerm used in those tests is stack allocated so we don't need to free it.
+}
+
+#[test]
+fn test_encode_raw_doc_ids_only() {
+    // Test cases for the raw doc ids encoder and decoder.
+    let tests = [
+        // (delta, expected encoding - raw 4-byte little-endian)
+        (0, vec![0, 0, 0, 0]),
+        (10, vec![10, 0, 0, 0]),
+        (256, vec![0, 1, 0, 0]),
+        (65536, vec![0, 0, 1, 0]),
+        (u16::MAX as u32, vec![255, 255, 0, 0]),
+        (u32::MAX, vec![255, 255, 255, 255]),
```

Comment:
Can we use `to_ne_bytes` for a more portable validation? Not that we really support big endian, so it's OK if not. Just wondering

Reply:
Anyway, let's not complicate the tests for no good reason

Reply:
Sure it will. 

It's really just a matter if the binary format should be endian dependent or not. But as you said, as look as this is only used an internal local representation it does not really matter.